tests/functional/p/protocol_classes_abstract.py

```diff
@@ -0,0 +1,36 @@
+"""Test that classes inheriting from protocols should not warn about abstract-method."""
+# pylint: disable=too-few-public-methods,disallowed-name,invalid-name
+from abc import abstractmethod
+from typing import Protocol, Literal
+
+
+class FooProtocol(Protocol):
+    """Foo Protocol"""
+
+    @abstractmethod
+    def foo(self) -> Literal["foo"]:
+        """foo method"""
+
+    def foo_no_abstract(self) -> Literal["foo"]:
+        """foo not abstract method"""
+
+class BarProtocol(Protocol):
+    """Bar Protocol"""
+    @abstractmethod
+    def bar(self) -> Literal["bar"]:
+        """bar method"""
+
+
+
+class FooBarProtocol(FooProtocol, BarProtocol, Protocol):
```

Comment:
Same, that PEP is quite over my head in its wording. I will add a test though!

Reply:
See `IndirectProtocol` below. Interesting PyCharm tells us this

![Screen Shot 2022-11-25 at 7 59 36 AM](https://user-images.githubusercontent.com/112832187/203970371-28538564-5381-46fe-8400-240dbea5b841.png)

Reply:
I do indeed think we should raise there as well. Since the class is not abstract or doesn't subclass `Protocol` it should implement all methods that are abstract.

A test where `IndirectProtocol` does subclass `abc.ABCMeta` and doesn't implement the methods but doesn't raise the message would also be good.

I think that is how I would interpret the PEP and PyCharm seems to do the same.